### PR TITLE
feat(auth): use new firebase login methods

### PR DESCRIPTION
**Pull request by kadikraman**
### Description

As of [react-native-firebase v3.2.3](https://github.com/invertase/react-native-firebase/releases/tag/v3.2.3), `signInWithCustomToken` is deprecated in favour of `signInAndRetrieveDataWithCustomToken`.

screen shot 2018-06-06 at 13 50 59

Found a similar issue [here](https://github.com/prescottprue/react-redux-firebase/issues/467) and implemented a fallback for users in the older version of firebase as suggested.

### Check List
- [ ] All tests passing
- [x] Docs updated with any changes or examples if applicable
- [ ] Added tests to ensure new feature(s) work properly

Fixes https://github.com/prescottprue/react-redux-firebase/issues/467

### CODE REVIEW

File: src/actions/auth.js [hunk 1]
```diff
@@ -443,21 +443,35 @@ export const login = (dispatch, firebase, credentials) => {
...
 
-  return firebase
-    .auth()
-    [method](...params)
+  const firebaseAuth = firebase.auth()
+
+  let firebaseMethod = firebaseAuth[method]
+
+  // support older versions of firebase that still use signInWithCustomToken & signInWithEmailAndPassword
+  switch (method) {
```
Comment (prescottprue) [inline src/actions/auth.js]: Is it possible for us to fit this logic in the `getLoginMethodAndParams` function?

File: test/utils.js [hunk 2]
```diff
@@ -129,7 +129,7 @@ export const fakeFirebase = {
               email: 'test@test.com',
               providerData: [{}]
             }),
-    signInWithCustomToken: () => {
+    signInAndRetrieveDataWithCustomToken: () => {
```
Comment (prescottprue) [inline test/utils.js]: Nice work updating the tests too!

**Comment by kadikraman** [comment]
@prescottprue thanks for the feedback. Updated to move logic to `getLoginMethodAndParams` which incidentally also fixed the tests that were failing before.

I think this definitely needs a test, but I'm not sure how to test for different versions of firebase. Any ideas?

**Comment by prescottprue** [comment]
@kadikraman Thanks for making the changes, and nice work!. I can look into good ways to test the different versions.

**Pull request merged**